crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -793,6 +793,39 @@ impl Starknet {
         }
     }
 
+    pub fn update_gas(&mut self, gas_prices: GasUpdate) -> DevnetResult<GasUpdate> {
+        // BlockContext needs to be reinitialized
+        self.block_context = Starknet::init_block_context(
+            gas_prices.gas_price_wei,
+            gas_prices.gas_price_strk,
+            gas_prices.data_gas_price_wei,
+            gas_prices.data_gas_price_strk,
+            constants::ETH_ERC20_CONTRACT_ADDRESS,
+            constants::STRK_ERC20_CONTRACT_ADDRESS,
+            ChainId::from(self.block_context.chain_info().chain_id.clone()),
+            self.block_context.block_info().block_number.0,
+        );
+
+        // Pending block header gas data needs to be updated
+        self.blocks.pending_block.header.l1_gas_price.price_in_wei =
+            GasPrice(u128::from(gas_prices.gas_price_wei));
+        self.blocks.pending_block.header.l1_data_gas_price.price_in_wei =
+            GasPrice(u128::from(gas_prices.data_gas_price_wei));
+        self.blocks.pending_block.header.l1_gas_price.price_in_fri =
+            GasPrice(u128::from(gas_prices.gas_price_strk));
+        self.blocks.pending_block.header.l1_data_gas_price.price_in_fri =
+            GasPrice(u128::from(gas_prices.data_gas_price_strk));
+
+        let gas_prices = self.block_context.block_info().gas_prices.clone();
+
```

Comment:
This line works, also another approach is a reference `&self.block_context.block_info().gas_prices;`

Reply:
Thanks! Fixed in 335c81e6457338806f4305b0e3b95c358cedb937